opnode/test/setup.go

```diff
@@ -288,12 +319,57 @@ func (cfg SystemConfig) start() (*System, error) {
 	}
 
 	sys.cfg.RollupConfig.Genesis = sys.RolupGenesis
-	sys.cfg.RollupConfig.BatchSenderAddress = submitterAddress
+	sys.cfg.RollupConfig.BatchSenderAddress = batchSubmitterAddr
+
+	// Deploy Deposit Contract
+	deployerPrivKey, err := sys.wallet.PrivateKey(accounts.Account{
+		URL: accounts.URL{
+			Path: sys.cfg.DeployerHDPath,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	opts, err := bind.NewKeyedTransactorWithChainID(deployerPrivKey, cfg.L1ChainID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Deploy contracts
+	sys.L2OOContractAddr, _, _, err = l2oo.DeployL2OutputOracle(
+		opts,
+		l1Client,
+		sys.cfg.L2OOCfg.SubmissionFrequency,
+		sys.cfg.L2OOCfg.L2BlockTime,
+		sys.cfg.L2OOCfg.GenesisL2Output,
+		sys.cfg.L2OOCfg.HistoricalTotalBlocks,
```

Comment:
This contract takes the starting block timestamp as a parameter as well.

Reply:
Nice catch. Will add